doc/dev/debugging.rst

```diff
@@ -403,4 +403,42 @@ The recompilation starts after the given checkpoint stage.
 
     replace_ir(f, "HLOLoweringPass", new_ir)
     f(2.0)
-    >> 8.0
\ No newline at end of file
+
+C Executable Generation
+=================
+
+Catalyst provides a way to generate a c executable that calls the shared object of the given
+qjit-decorated function.
+In ``catalyst.debug.compiler_functions``, ``compile_executable(fn: QJIT, *args) -> str, str`` takes
+a qjit-decorated function and arguments for that function.
```

Comment:
```suggestion
Let's use cross references :)
```suggestion
The function :func:`~.debug.compile_executable` takes a qjit-decorated function and
concrete arguments to that function as input.
```
```